python/hail/dataset.py

```diff
@@ -164,6 +164,77 @@ def file_version(self):
 
         return self._jvds.fileVersion()
 
+    @handle_py4j
+    @record_method
+    @typecheck_method(split_variant_expr=oneof(strlike, listof(strlike)),
+                      split_genotype_expr=oneof(strlike, listof(strlike)),
+                      variant_expr=oneof(strlike, listof(strlike)))
+    def annotate_alleles_expr_generic(self, split_variant_expr, split_genotype_expr, variant_expr):
+        """Annotate alleles with expression.
+
+        ``annotate_alleles_expr`` works by splitting variants into
+        biallelics using ``split_genotype_expr``, annotating those
+        biallelic variants with ``variant_expr``, and then collecting
+        the biallelic variant annotations into an array and annotating
+        the original variants with that allele-indexed array.
+
+        ``split_variant_expr`` is an annotation expression to update
+        the split variant annotations for each outupt biallelic
+        variant.  It updates ``va`` and is evaluated in the following
+        namespace:
+
+        - ``global``: global annotations
+        - ``v``: :ref:`variant(GR)`: the old variant
+        - ``newV``: :ref:`variant(GR)`: the new, split variant
+        - ``va``: the old variant annotations
+        - ``aIndex``: the index of the input allele to the output variant
+        - ``wasSplit``: True if the original variant was multiallelic
+
+        ``split_genotype_expr`` is an annotation expression to update
+        the split genotype annotations for each output biallelic
+        variant.  It updates `g` and is evaluated in the following
+        namespace:
+
```

Comment:
Worth putting a couple of commonly-used examples in? Particularly here, but perhaps also for the others.